nvm.sh

```diff
@@ -695,6 +695,82 @@ nvm_binary_available() {
   nvm_version_greater_than_or_equal_to "$(nvm_strip_iojs_prefix "${1-}")" v0.8.6
 }
 
+nvm_set_colors() {
+  if [ nvm_has_colors ]
+  then
+    COLORS=$1
+    INSTALLED_COLOR="$(nvm_print_color_code ${COLORS:0:1})"
+    LTS_AND_SYSTEM_COLOR="$(nvm_print_color_code ${COLORS:1:1})"
+    CURRENT_COLOR="$(nvm_print_color_code ${COLORS:2:1})"
+    NOT_INSTALLED_COLOR="$(nvm_print_color_code ${COLORS:3:1})"
+    DEFAULT_COLOR="$(nvm_print_color_code ${COLORS:4:1})"
+    printf "Setting colors to: \033[${INSTALLED_COLOR}${COLORS:0:1}\033[${LTS_AND_SYSTEM_COLOR}${COLORS:1:1}\033[${CURRENT_COLOR}${COLORS:2:1}\033[${NOT_INSTALLED_COLOR}${COLORS:3:1}\033[${DEFAULT_COLOR}${COLORS:4:1}\033[0m"
+    export NVM_COLORS="$INSTALLED_COLOR:$LTS_AND_SYSTEM_COLOR:$CURRENT_COLOR:$NOT_INSTALLED_COLOR:$DEFAULT_COLOR"
+  fi
+}
+
+nvm_get_colors() {
+  if [ $NVM_COLORS ]; then
+    INSTALLED_COLOR="$(cut -d':' -f1 <<<$NVM_COLORS)"
+    SYSTEM_COLOR="$(cut -d':' -f2 <<<$NVM_COLORS)"
+    CURRENT_COLOR="$(cut -d':' -f3 <<<$NVM_COLORS)"
+    NOT_INSTALLED_COLOR="$(cut -d':' -f4 <<<$NVM_COLORS)"
+    DEFAULT_COLOR="$(cut -d':' -f5 <<<$NVM_COLORS)"
+    LTS_COLOR="$(echo $SYSTEM_COLOR | tr '0;' '1;')"
+  else 
+    CURRENT_COLOR="0;32m"
+    INSTALLED_COLOR="0;34m"
+    LTS_COLOR="1;33m"
+    NOT_INSTALLED_COLOR="0;31m"
+    DEFAULT_COLOR="0;37m"
+    SYSTEM_COLOR="0;33m"
+  fi
+}
+
+nvm_print_color_code(){
+  case $1 in
```

Comment:
```suggestion
  case "${1}" in
```
this also needs to exit nonzero if the first argument is not provided, and also in the `*` case. there should also be unit tests for this function.

Reply:
Hi @ljharb, is it standard practice to add a * case if we are only calling this function from another function if the input matches our specified conditions, and therefore we don't expect to get bad data passed into it?

Reply:
Adding a `*` case that exits 1.